sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/splittabledofn/WatermarkEstimator.java

```diff
@@ -44,4 +45,19 @@
    * <p>The state returned must not be mutated.
    */
   WatermarkEstimatorStateT getState();
+
+  /**
+   * Validates that a given watermark is within timestamp min and max bounds.
+   *
+   * @param watermark watermark to validate
+   */
+  static void ensureWatermarkWithinBounds(Instant watermark) {
```

Comment:
It makes more sense to add this method to BoundedWindow as it should apply to a lot more places then just WatermarkEstimators.

Reply:
Done.